modules/storage-api/src/testFixtures/java/org/apache/ignite/internal/storage/impl/TestMvPartitionStorage.java

```diff
@@ -426,6 +448,51 @@ public ReadResult next() {
         return map.ceilingKey(lowerBound);
     }
 
+    @Override
+    public @Nullable BinaryRowWithRowId pollForVacuum(HybridTimestamp lowWatermark) {
+        Iterator<IgniteBiTuple<VersionChain, RowId>> it = gcQueue.iterator();
```

Comment:
No synchronization whatsoever. How about changing `gcQueue` to `ConcurrentSkipListSet`?

Reply:
Sure, thank you!